sumologic/resource_sumologic_monitors_library_monitor.go

```diff
@@ -305,19 +307,29 @@ func resourceSumologicMonitorsLibraryMonitorRead(d *schema.ResourceData, meta in
 		schemaInternalNotification := make([]interface{}, 1)
 		internalNotification := make(map[string]interface{})
 		internalNotificationDict := n.Notification.(map[string]interface{})
+		// log.Printf("monitor.Notification %v", n.Notification)
 		if internalNotificationDict["connectionType"] != nil {
 			internalNotification["connection_type"] = internalNotificationDict["connectionType"].(string)
+		} else {
+			// for backwards compatibility
+			internalNotification["connection_type"] = internalNotificationDict["actionType"].(string)
+			// convert from old action_type name to new connection_type name if applicable
+			if internalNotification["connection_type"].(string) == "EmailAction" {
+				internalNotification["connection_type"] = "Email"
+			}
+			if internalNotification["connection_type"].(string) == "NamedConnectionAction" {
+				internalNotification["connection_type"] = "Webhook"
```

Comment:
This is not correct. `NamedConnectionAction` can be of type `Slack`, `Jira`, `PagerDuty`, `Webhook` etc.

Reply:
This is specifically for the case where we have existing `action_type = "NamedConnectionAction"` - in that case we don't have enough information from the resource itself to decide which of the webhook types it should be so only the generic "Webhook" connection type makes sense right? if the user has `action_type = "PagerDuty"` then connection_type will be "PagerDuty"

Reply:
`NamedConnectionAction` allows all types right now. The good part is that we have done one round of migration and the next one is planned for soon so we are fine at DB layer. But if a customer has set up a monitor with this type for `slack`/`pagerduty` etc via Terraform, it will save it as `Webhook` on update after this change. There is no clean solution to make this truly backward compatible. We will have to work with customers to use `Slack` etc instead of `NamedConnectionAction`.

Reply:
Yup - also from what I tested, there doesn't seem to be an immediate functional impact to using "Webhook" for connection_type instead of the more specific types "PagerDuty"/"Slack" - the notification/payload still seems to shows up correctly in the UI and API